lib/charliecloud.py

```diff
@@ -446,6 +447,18 @@ def set_(dst_key, *src_keys):
             except AttributeError:
                FATAL("can't parse config: bad Env line: %s" % line)
             self.metadata["env"][k] = v
+      # History. Set the empty_layer value of each non-final history entry to
+      # true.
+      try:
+         hist = config["history"]
```

Comment:
I'd suggest an explicit translation step here. I.e., do we really want all fields in each history entry from the config? Which ones do we care about and which do we not?

Reply:
@heasterday argues that something generic like `FROM TAG` in leou of the base image's actual history isn't very helpful; the tag TAG referenced could be from a locally built image and unavailable for inspection from a registry like Dockerhub.

Determining what is useful and what isn't seems strange for the base image. I personally prefer to replace it with something simple like `created_by: FROM TAG` OR pass all the base image history unedited aside from the `empty_layer` hack.